src/lib/SmartTokens/Contracts/Issuance.hs

```diff
@@ -88,7 +90,7 @@ mkProgrammableLogicMinting :: ClosedTerm (PAsData PCredential :--> PAsData PCurr
 mkProgrammableLogicMinting = plam $ \programmableLogicBase nodeCS mintingLogicCred ctx -> P.do
   ctxF <- pletFields @'["txInfo", "redeemer", "scriptInfo"] ctx
   infoF <- pletFields @'["referenceInputs", "outputs", "mint", "wdrl"] ctxF.txInfo
-  let red = punsafeCoerce @_ @_ @PSmartTokenMintingAction (pto ctxF.redeemer)
+  let red = punsafeCoerce @_ @_ @(PAsData PSmartTokenMintingAction) (pto ctxF.redeemer)
```

Comment:
Bug 1. This type is actually data encoded but it was unsafeCoerced into non-data encoding.